[CIVIS-9838] Add docs for AWS CodePipeline log correlation (#22807)
DataDog/documentation: content/en/continuous_integration/pipelines/awscodepipeline.md

| Pipeline Visibility | Platform | Definition |
|---|---|---|
| [Partial retries][14] | Partial pipelines | View partially retried pipeline executions. |
| *[Running pipelines][15] | Running pipelines | View pipeline executions that are running. |

> | **Logs correlation | Logs correlation	| Correlate pipeline and job spans to logs and enable [job log correlation][16]. |

Reviewer: ```suggestion
| **Logs correlation | Logs correlation	| [Enable job log correlation](#enable-log-correlation) to correlate pipeline and job spans to logs. |
```

Author: We use the `Correlate pipeline and job spans to logs and enable job log correlation` in all the docs for other providers - imho we should be consistent.

*AWS CodePipeline running pipelines don't have Git information until they have finished.\

> **AWS CodePipeline logs correlation is currently only available for AWS CodeBuild actions.

Reviewer: ```suggestion
**AWS CodePipeline logs correlation is only available for AWS CodeBuild actions.
```

The steps above allow you to add the pipeline execution ID to your CodeBuild action environment variables. For more information on working with variables, see the [official AWS guide][10].

### Enable log correlation

> The AWS CodePipeline integration supports correlating **CodeBuild** job's actions with their respective job and pipeline spans. To enable log collection for your CodeBuild actions, follow the steps in the general [AWS logs forwarding guide][17].

Reviewer: ```suggestion
The AWS CodePipeline integration supports correlating **CodeBuild** job actions with their respective job and pipeline spans. To enable log collection for your CodeBuild actions, follow the steps in the [AWS log forwarding guide][16].
```

The steps above allow you to add the pipeline execution ID to your CodeBuild action environment variables. For more information on working with variables, see the [official AWS guide][10].

### Enable log correlation

> The AWS CodePipeline integration supports correlating **CodeBuild** actions with their respective job and pipeline spans. To enable log collection for your CodeBuild actions, follow the steps in the [AWS log forwarding guide][16].

Reviewer: ```suggestion
The AWS CodePipeline integration supports correlating **CodeBuild** actions with their respective job and pipeline spans. To enable log collection for your CodeBuild actions, see [Enable logging for your AWS service][16] in the AWS log forwarding guide.
```

[12]: https://app.datadoghq.com/ci/pipeline-executions
[13]: https://docs.datadoghq.com/continuous_integration/troubleshooting/#the-default-branch-is-not-correct
[14]: /glossary/#partial-retry
[15]: /glossary/#running-pipeline

> [16]: /continuous_integration/pipelines/awscodepipeline/#enable-log-correlation
> [17]: /logs/guide/send-aws-services-logs-with-the-datadog-lambda-function

Reviewer: ```suggestion
[16]: /logs/guide/send-aws-services-logs-with-the-datadog-lambda-function
```
